### DISCUSSION TITLE
MaterialId is always -1 even if the textures are present

### DISCUSSION THREAD

**Discussion by daker** [syoyo/tinyusdz#205] [General]
Hi,

Thank you for putting the effort on the library, i am using the wasm version and facing an issue where for **some models** provided by Apple, the MaterialId is always -1 which result in no texture displayed.

Do you have an idea what causes the issue ?

https://developer.apple.com/augmented-reality/quick-look/models/stratocaster/fender_stratocaster.usdz
https://developer.apple.com/augmented-reality/quick-look/models/drummertoy/toy_drummer_idle.usdz
https://developer.apple.com/augmented-reality/quick-look/models/biplane/toy_biplane_idle.usdz
https://developer.apple.com/augmented-reality/quick-look/models/vintagerobot2k/robot_walk_idle.usdz

![image](https://github.com/user-attachments/assets/8d2d8205-9ec3-4679-89b9-178b4e8abc23)

![image](https://github.com/user-attachments/assets/824a830d-d2cd-49f6-9294-ab2711bcc9ee)

![image](https://github.com/user-attachments/assets/450a63b3-2457-4c21-84e5-e4a78cd06982)

**Comment by syoyo** [comment]
You should post reproducible code.

**Reply by daker** [reply]
i did left a reproduction code

https://github.com/lighttransport/tinyusdz/discussions/205#discussioncomment-11101613

**Comment by daker** [comment]
Here is a simple reproduction code with the model, both meshes from the .usdz have materials but the code return -1

[tinyusdz-bug.zip](https://github.com/user-attachments/files/17574988/tinyusdz-bug.zip)

**Comment by syoyo** [comment]
You can try adding 'converter.GetWarning()' and how it goes. It shows warning message during scene conversion

https://github.com/lighttransport/tinyusdz/blob/db4e6aa1b15c07a1941b1ab53b33d8a11c946bde/web/binding.cc#L103

**Reply by daker** [reply]
Unfortunately i get no warnings, any other ideas to debug this issue ?

```cpp
loaded_ = converter.ConvertToRenderScene(env, &render_scene_);
warn_ = converter.GetWarning();

const std::string warn() const { return warn_; }

 .function("warn", &TinyUSDZLoader::warn)
```

**Comment by syoyo** [comment]
You can debug the C++ part of TinyUSDZ by:

- add printf to suspicious place
- enable TINYUSDZ_DEBUG_PRINT https://github.com/lighttransport/tinyusdz/blob/1b3b08f102ee2882e36e17f342aef40381c4ec47/CMakeLists.txt#L172
